src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallSummaryGenerator.kt

```diff
@@ -27,39 +31,57 @@ class RecallSummaryGenerator(
-      setVariable("furtherCharge", context.recall.reasonsForRecall.contains(ReasonForRecall.ELM_FURTHER_OFFENCE) || context.recall.reasonsForRecall.contains(ReasonForRecall.POOR_BEHAVIOUR_FURTHER_OFFENCE))
+      val recall = context.recall
+      with(recall.mappaLevel) {
+        setVariable("mappaLevel1", this?.equals(LEVEL_1) ?: false)
+        setVariable("mappaLevel2", this?.equals(LEVEL_2) ?: false)
+        setVariable("mappaLevel3", this?.equals(LEVEL_3) ?: false)
+      }
+
+      with(recall.sentencingInfo) {
+        setVariable("lengthOfSentence", this?.sentenceLength?.prettyPrint())
+        setVariable("indexOffence", this?.indexOffence)
+        setVariable("sentencingCourt", this?.sentencingCourt)
+        setVariable("sentencingDate", this?.sentenceDate?.format(dateFormatter))
+        setVariable("sed", this?.sentenceExpiryDate?.format(dateFormatter))
+      }
+
+      with(recall.probationInfo) {
+        setVariable("offenderManagerName", this?.probationOfficerName)
+        setVariable("offenderManagerContactNumber", this?.probationOfficerPhoneNumber)
+      }
 
-      setVariable("pncCroNumber", context.prisoner.croNumber)
-      setVariable("offenderManagerName", context.recall.probationInfo?.probationOfficerName)
-      setVariable("offenderManagerContactNumber", context.recall.probationInfo?.probationOfficerPhoneNumber)
-      setVariable("policeSpoc", context.recall.localPoliceForce)
+      setVariable("policeFileName", recall.previousConvictionMainName)
+      setVariable("prisonNumber", recall.bookingNumber)
+      setVariable("pnomisNumber", recall.nomsNumber)
+      setVariable("releaseDate", recall.lastReleaseDate?.format(dateFormatter))
+      setVariable("furtherCharge", recall.hasFurtherCharge())
+      setVariable("policeSpoc", recall.localPoliceForce)
       setVariable("currentPrison", context.currentPrisonName)
-      setVariable("sentencingCourt", context.recall.sentencingInfo?.sentencingCourt)
-      setVariable("sentencingDate", context.recall.sentencingInfo?.sentenceDate?.format(dateFormatter))
-      setVariable("sed", context.recall.sentencingInfo?.sentenceExpiryDate?.format(dateFormatter))
-      setVariable("vulnerabilityDetail", context.recall.vulnerabilityDiversityDetail ?: "None")
-      setVariable("contraband", if (context.recall.contrabandDetail.isNullOrEmpty()) "NO" else "YES")
-      setVariable("contrabandDetail", context.recall.contrabandDetail)
+      setVariable("vulnerabilityDetail", recall.vulnerabilityDiversityDetail ?: "None")
+      setVariable("contraband", if (recall.contrabandDetail.isNullOrEmpty()) "NO" else "YES")
+      setVariable("contrabandDetail", recall.contrabandDetail)
+      setVariable("releasingPrison", context.lastReleasePrisonName)
+
       setVariable("logoFileName", RecallSummaryLogo.fileName)
     }.let {
       templateEngine.process("recall-summary", it)
     }
+
+  private fun Recall.hasFurtherCharge() =
```

Comment:
How about moving this definition onto Recall itself?

Reply:
Could do if it was ever needed anywhere else